packages/flutter/lib/src/material/app_bar_theme.dart

```diff
@@ -70,27 +135,35 @@ class AppBarTheme with Diagnosticable {
   ///
   ///  * [backgroundColor], which overrides the default value of
   ///    [AppBar.backgroundColor] in all descendant [AppBar] widgets.
-  final Color? foregroundColor;
+  Color? get foregroundColor => _data != null ? _data.foregroundColor : _foregroundColor;
+
+  /// Overrides the default value of [AppBar.color] in all
+  /// descendant [AppBar] widgets.
+  ///
+  /// This property is obsolete and will be deprecated in a future release:
+  /// please use [backgroundColor] instead.
+  Color? get color => _data != null ? _data.color : _backgroundColor;
```

Comment:
I think we don't need to add getter for `color`. It only exist as a constructor parameter previously.